### ISSUE TITLE
Twitter is missing in OAuth2

### ISSUE THREAD

**Issue by jadwigo** [bolt/ClientLogin#13]
If you enable twitter the clientlogin will default to OAuth2 on line: https://github.com/GawainLynch/bolt-extension-clientlogin/blob/master/src/Session.php#L401

This won't work and will return 

```
Bolt - Kritieke fout.
Class Bolt\Extension\Bolt\ClientLogin\Exception\ProviderException
Bericht:
Code 0

Bolt\Extension\Bolt\ClientLogin\Session->setProvider()
[root]extensions/vendor/bolt/clientlogin/src/Session.php, regel 170

Bolt\Extension\Bolt\ClientLogin\Session->loginOAuth()
[root]extensions/vendor/bolt/clientlogin/src/Session.php, regel 95

Bolt\Extension\Bolt\ClientLogin\Session->doLogin()
[root]extensions/vendor/bolt/clientlogin/src/Controller/ClientLoginController.php, regel 69

Symfony\Component\HttpKernel\HttpKernel->handleRaw()
[root]vendor/symfony/http-kernel/Symfony/Component/HttpKernel/HttpKernel.php, regel 66

Symfony\Component\HttpKernel\HttpKernel->handle()
[root]vendor/silex/silex/src/Silex/Application.php, regel 581

Silex\Application->handle()
[root]vendor/silex/silex/src/Silex/Application.php, regel 558

Silex\Application->run()
[root]index.php, regel 49
```

Because the loginprovider for Twitter is not present (there is something in OAuth1 but that will probably work differently.

Version: Bolt 2.2.1 with clientlogin dev-master

**Comment by GawainLynch** [comment]
Yep… Twitter doesn't support OAuth2

**Comment by jadwigo** [comment]
Which should either be fixed by switching to oauth1 in that case or removing it from the readme?

**Comment by GawainLynch** [comment]
Haha… Sorry, too quick on the send button while debugging (interestingly the Bolt 2.3 authentication workflow/security).

Yes, the intent was to incorporate the OAuth1 library, once they updated their Guzzle requirement to 5, or I found time to port it myself.  

The further intent is to move the OAuth2 library to the 1.0 version when that is stable/released. At present I have ported it to Guzzle 5, but its fiddly.
